### PR TITLE
chore: Run tests in development mode by default

### PR DESCRIPTION
**Pull request by Artur-**
Production mode can be enabled using -Pproduction

### CODE REVIEW

File: packages/java/tests/pom.xml [hunk 1]
```diff
@@ -1,288 +1,311 @@
...
+                <plugin>
+                    <artifactId>maven-clean-plugin</artifactId>
+                    <version>3.2.0</version>
+                    <configuration>
+                        <filesets>
+                            <fileset>
+                                <directory>${project.basedir}</directory>
+                                <includes>
+                                    <include>node_modules/**</include>
+                                    <include>types.d.ts</include>
```
Comment (taefi) [inline packages/java/tests/pom.xml:170]: Is the default `types.d.ts` file generated by flow working for React apps nowadays?

File: packages/java/tests/pom.xml [hunk 2]
```diff
@@ -1,288 +1,311 @@
...
+                <pluginManagement>
+                    <plugins>
+                        <plugin>
+                            <groupId>dev.hilla</groupId>
+                            <artifactId>hilla-maven-plugin</artifactId>
+                            <version>${project.version}</version>
+                            <executions>
+                                <execution>
+                                    <goals>
+                                        <goal>prepare-frontend</goal>
```
Comment (taefi) [inline packages/java/tests/pom.xml:263]:
```suggestion
```

**Review by taefi — CHANGES_REQUESTED** [review]

File: packages/java/tests/pom.xml (hunk 2 above)
Comment (Artur-) [reply packages/java/tests/pom.xml]: Done

File: packages/java/tests/pom.xml (hunk 1 above)
Comment (Artur-) [reply packages/java/tests/pom.xml:170]: Probably not but this part is not changed in this PR

File: .github/workflows/validation.yml [hunk 3]
```diff
@@ -199,6 +199,7 @@ jobs:
               -Dmaven.wagon.http.retryHandler.count=3 \
               -P\!it-modules \
               -Pit-modules-${{ matrix.it-modules }} \
+              -Pproduction \
```
Comment (taefi) [inline .github/workflows/validation.yml:202]: Is the PR title wrong or this profile activation?

**Review by taefi — CHANGES_REQUESTED** [review]

File: .github/workflows/validation.yml (hunk 3 above)
Comment (Artur-) [reply .github/workflows/validation.yml:202]: Well the intent is to run them in dev mode when you run locally, so you can develop, and in production mode in CI so it is faster

**Review by taefi — APPROVED** [review]

**Review by taefi — APPROVED** [review]

**Comment by Artur-** [comment]
The validation build will not pass here as it uses the old configuration

**Pull request merged**